src/utils/aadV2TokenProvider.ts

```diff
@@ -0,0 +1,269 @@
+import { Clipboard, commands, env, Uri, window } from 'vscode';
+import * as Constants from '../common/constants';
+import { HttpRequest } from '../models/httpRequest';
+import { HttpClient } from './httpClient';
+import { EnvironmentVariableProvider } from './httpVariableProviders/environmentVariableProvider';
+
+/*
+ AppId provisioned to allow users to explicitly consent to permissions that this app can call
+*/
+const AadV2TokenProviderClientId = "07f0a107-95c1-41ad-8f13-912eab68b93f";
+
+export class AadV2TokenProvider {
+    private readonly _httpClient: HttpClient;
+    private readonly clipboard: Clipboard;
+
+    public constructor() {
+        this._httpClient = new HttpClient();
+        this.clipboard = env.clipboard;
+    }
+
+    public async acquireToken(name: string): Promise<string> {
+
+        const authParams = new AuthParameters();
+        await authParams.parseName(name);
```

Comment:
Why not directly provide a static method named `parseName` of class `AuthParameters` to generate an `AuthParameters` instance like following:
```typescript
const authParams =  AuthParameters.parse(name);
```

Reply:
That works too.  Changed.